codec-http/src/main/java/io/netty/handler/codec/http/multipart/DeleteFileOnExitHook.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http.multipart;
+
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Extend {@link java.io.DeleteOnExitHook} to dynamically manipulate hook records.
+ */
+public final class DeleteFileOnExitHook {
+    private static Set<String> files = new HashSet<String>();
+    static final InternalLogger logger = InternalLoggerFactory.getInstance(DeleteFileOnExitHook.class);
+
+    private DeleteFileOnExitHook() {
+    }
+
+    static {
+        // DeleteOnExitHook must be the last shutdown hook to be invoked.
+        // Application shutdown hooks may add the first file to the
+        // delete on exit list and cause the DeleteOnExitHook to be
+        // registered during shutdown in progress.
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+
+            @Override
+            public void run() {
+                runHooks();
+            }
+        });
+    }
+
+    /**
+     * Remove from the pool to reduce space footprint.
+     *
+     * @param file tmp file path
+     */
+    public static synchronized void remove(String file) {
+        try {
+            if (files != null) {
+                files.remove(file);
+            }
+        } catch (Exception e) {
+            logger.warn("The cleanup file path failed.", e);
+        }
```

Comment:
I don't think there's any exception that can happen here?

Reply:
Maybe it is redundant，If an unknown problem occurs with `add` and `remove`, I just want to make sure that it will not affect the normal process. 

Or am I thinking a little too much?

Reply:
Yeah, they're just methods on Set. They can only throw Errors or RuntimeExceptions, and I think we should just let those bubble out if they ever happen.

Reply:
I agree with @chrisvest here..

Reply:
Okay, I've adjusted it.